### PR TITLE
Add sorin theme

### PR DESCRIPTION
**Pull request by pickfire**
Hi,
  This is my first pull request. Take a look at #2100

Thanks.

### CODE REVIEW

**Comment by ridiculousfish** [comment]
Thanks, I'll take a look!

**Comment by ridiculousfish** [comment]
Merged as a150ae76ac15ced7965f909e6b4b433c57321d84. Thanks for the nice contribution!

File: share/tools/web_config/sample_prompts/sorin.fish [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# Name:   Sorin
+# Author: Ivan Tham <ivanthamjunhoe@gmail.com>
+
+# If login shell, don't do anything
+if status --is-login; and function fish_prompt; echo -n '% '; end; end
```
Comment (zanchey) [inline share/tools/web_config/sample_prompts/sorin.fish]: Why does this prompt treat login shells differently?

File: share/tools/web_config/sample_prompts/sorin.fish [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+function fish_prompt
+	if not set -q -g __prompt_sorin_functions-defined
+		set -g __prompt_sorin_functions-defined
+	end
```
Comment (zanchey) [inline share/tools/web_config/sample_prompts/sorin.fish]: Is this needed? I can't see that it gets tested anywhere.

File: share/tools/web_config/sample_prompts/sorin.fish [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+	test $SSH_TTY; and printf (set_color red)(whoami)(set_color white)'@'(set_color yellow)(hostname)' '
```
Comment (zanchey) [inline share/tools/web_config/sample_prompts/sorin.fish]: This probably should be `test -n "$SSH_TTY"` or similar.

File: share/tools/web_config/sample_prompts/sorin.fish (hunk 1 above)
Comment (pickfire) [inline share/tools/web_config/sample_prompts/sorin.fish]: It prevent errors for unsupportable characters in the linux framebuffer (**Ctrl+Alt+F2**).

File: share/tools/web_config/sample_prompts/sorin.fish (hunk 2 above)
Comment (pickfire) [inline share/tools/web_config/sample_prompts/sorin.fish]: I doesn't know that does line **7-10** do to, I had copied from other prompts.

File: share/tools/web_config/sample_prompts/sorin.fish (hunk 3 above)
Comment (pickfire) [inline share/tools/web_config/sample_prompts/sorin.fish]: I want that **error** returned for **`and`** to execute the code when it is *true*.

File: share/tools/web_config/sample_prompts/sorin.fish (hunk 1 above)
Comment (faho) [inline share/tools/web_config/sample_prompts/sorin.fish]: Then shouldn't that be tested via "test $TERM = Linux"?

Login shells can also be used in other contexts, and there can also be non-login shells in a framebuffer VT.

Comment (pickfire) [inline share/tools/web_config/sample_prompts/sorin.fish]: Sorry, I didn't notice that.

**Comment by pickfire** [comment]
I have some fixes in #2243.

**Pull request closed without merging**